openbas-api/src/test/java/io/openbas/rest/inject/InjectApiTest.java

```diff
@@ -948,4 +953,130 @@ void shouldComputeAgentStatusAsMayBePrevented() throws Exception {
+                  .forInjectStatus(InjectStatusFixture.createPendingInjectStatus())
+                  .withExecutionTraces(traces))
+          .persist()
+          .get();
+    }
+
+    private String performGetRequest(String baseUri, String injectId, String targetId)
+        throws Exception {
+      MockHttpServletRequestBuilder requestBuilder =
+          get(baseUri).accept(MediaType.APPLICATION_JSON).param("injectId", injectId);
+
+      if (targetId != null) {
+        requestBuilder.param("targetId", targetId);
+        requestBuilder.param("targetType", TargetType.AGENT.name());
+      }
+
+      return mvc.perform(requestBuilder)
+          .andExpect(status().is2xxSuccessful())
+          .andReturn()
+          .getResponse()
+          .getContentAsString();
+    }
+
+    @Test
+    @DisplayName("Fetch execution traces by target and inject")
+    void should_fetch_execution_traces_by_target_and_inject() throws Exception {
+      AgentComposer.Composer agent =
+          agentComposer.forAgent(AgentFixture.createDefaultAgentService());
+
+      endpointComposer.forEndpoint(EndpointFixture.createEndpoint()).withAgent(agent).persist();
+
+      Inject inject =
+          buildInjectWithTraces(
+              List.of(
+                  executionTraceComposer
+                      .forExecutionTrace(ExecutionTraceFixture.createDefaultExecutionTraceStart())
+                      .withAgent(agent),
+                  executionTraceComposer
+                      .forExecutionTrace(
+                          ExecutionTraceFixture.createDefaultExecutionTraceComplete())
+                      .withAgent(agent)));
+
+      Agent savedAgent = inject.getStatus().get().getTraces().get(0).getAgent();
+      String response =
+          performGetRequest(INJECT_URI + "/execution-traces", inject.getId(), savedAgent.getId());
+
+      List<ExecutionTraceOutput> expected = new ArrayList<>();
+      expected.add(
+          ExecutionTraceOutput.builder()
+              .action(ExecutionTraceAction.START)
+              .message("Info")
+              .status(ExecutionTraceStatus.INFO)
+              .build());
+      expected.add(
+          ExecutionTraceOutput.builder()
+              .action(ExecutionTraceAction.COMPLETE)
+              .message("Success")
+              .status(ExecutionTraceStatus.SUCCESS)
+              .build());
+
+      assertThatJson(response)
+          .when(Option.IGNORING_ARRAY_ORDER)
+          .when(Option.IGNORING_EXTRA_FIELDS)
+          .whenIgnoringPaths(
+              "execution_time",
+              "execution_agent",
+              "execution_agent.agent_id",
+              "execution_agent.agent_privilege",
+              "execution_agent.agent_deployment_mode",
+              "execution_agent.agent_executed_by_user",
+              "execution_agent.agent_active",
+              "execution_agent.agent_last_seen")
+          .isEqualTo(mapper.writeValueAsString(expected));
```

Comment:
If you want to assert on a specific node, you can use `.node(string)` e.g. (do check the actual node name)
```suggestion
      assertThatJson(response)
          .when(Option.IGNORING_ARRAY_ORDER)
          .when(Option.IGNORING_EXTRA_FIELDS)
          .node("traces")
          .isEqualTo(mapper.writeValueAsString(expected));
```